src/main/java/vazkii/botania/common/brew/potion/PotionBloodthirst.java

```diff
@@ -26,19 +26,19 @@ public PotionBloodthirst() {
 		super(EffectType.BENEFICIAL, 0xC30000);
 	}
 
-	public static void onSpawn(LivingSpawnEvent.CheckSpawn event) {
-		if (event.getResult() != Event.Result.ALLOW && event.getEntityLiving() instanceof IMob) {
-			AxisAlignedBB aabb = new AxisAlignedBB(event.getX() - RANGE, event.getY() - RANGE, event.getZ() - RANGE,
-					event.getX() + RANGE, event.getY() + RANGE, event.getZ() + RANGE);
-			for (PlayerEntity player : event.getWorld().getPlayers()) {
+	// TODO: This runs a check for *every* player, would it be faster to just query players within the spawn AABB?
```

Comment:
No, because we used to do that and it was actually slower. Touching the chunks to query their entities for players has super high over head, while doing a distance check for at most 100 or so players is much cheaper.